capstone/src/main/java/com/google/sps/servlets/studentServlet/StudentServlet.java

```diff
@@ -0,0 +1,125 @@
+package com.google.sps.servlets;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns a student's profile content */
+@WebServlet("/student-data")
+public class StudentServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get student object based on the logged in email
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+
+    // Get the user's information from Datastore
+    Query query = new Query(userEmail);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    ImmutableList<Entity> students = ImmutableList.copyOf(results.asIterable());
+
+    // Add user to Datastore if this is the first time they login
+    if (students.isEmpty()) {
+      Entity studentEntity = createStudentEntity(userEmail);
+      datastore.put(studentEntity);
+      results = datastore.prepare(query);
+      students = ImmutableList.copyOf(results.asIterable());
+    }
+    // A user can only be logged in with one email address at a time
+    Entity currentStudent = students.get(0);
+
+    // Get club list from entity and convert to an ImmutableList
+    // Initally empty in case there is no club list
+    ImmutableList<String> clubs = ImmutableList.of();
+    if (currentStudent.getProperty(Constants.PROPERTY_CLUBS) != null) {
+      String clubsAsString = currentStudent.getProperty(Constants.PROPERTY_CLUBS).toString();
+      // Convert string representation of a list to an ImmutableList
+      clubs =
+          ImmutableList.copyOf(clubsAsString.substring(1, clubsAsString.length() - 1).split(","));
+    }
+
+    // Create Student object based on stored information
+    Student student =
+        new Student(
+            currentStudent.getProperty(Constants.PROPERTY_NAME).toString(),
+            Integer.parseInt(currentStudent.getProperty(Constants.PROPERTY_GRADYEAR).toString()),
```

Comment:
Is this the only way to get this property as an int? It seems kind of weird that you have to parse it to a string and then to an int. But maybe this is the only way possible, just wondering if you've looked into this?

Reply:
I've seen .intValue() used (https://www.geeksforgeeks.org/integer-intvalue-method-in-java/) but I believe it only works on Number objects. However, I get an error when I cast this property to Number so I'm assuming converting to String and then calling parseInt is generally how it is done? I'm not able to find another way to do it :(